internal/security/security_test.go

```diff
@@ -0,0 +1,145 @@
+package security
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/schema"
+	"github.com/stretchr/testify/assert"
+)
+
+// setMockPasswordPolicyConfig helper for returning customized mock config
+func setMockPasswordPolicyConfig(policyEnabled bool, authPolicyEnabled bool, authMinPasswordLength int,
```

Comment:
Nitpick, but the `true` and `12` in the tests are really hard to tell apart. I'd personally do something like this:

```suggestion
type mockPolicyOpts struct {
  policyEnabled, authPolicyEnabled, reqNumber, reqCase bool
  minPasswordLength, specialChars int
}
// setMockPasswordPolicyConfig helper for returning customized mock config
func setMockPasswordPolicyConfig(opts mockPolicyOpts)
```

and then you can leverage default values for struc fields (meaning: you don't have to initialize a `bool` to `false`) and don't need the local vars to describe integers. 

This

```go
setMockPasswordPolicyConfig(false, true, authPolicyLength, authPolicySpChr,
			false, false)
```

could then look like

```go
setMockPasswordPolicyConfig(mockPolicyOpts{authPolicyEnabled: true, minPasswordLength: 15, specialChars: 2})
```